app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -55,12 +58,39 @@ class ScanSimpleIdUpdate @Inject constructor(
     if (model.isSearching) return noChange()
 
     return try {
-      val bpPassportCode = UUID.fromString(event.text)
-      val identifier = Identifier(bpPassportCode.toString(), BpPassport)
-      next(model = model.searching(), SearchPatientByIdentifier(identifier))
-    } catch (e: IllegalArgumentException) {
-      crashReporter.report(e)
-      noChange()
+      val patientPrefillInfoDetail = parseJsonIntoObject(event.text)
```

Comment:
We are nesting try/catch blocks. Can we think of a way to simplify this, please?